src/ControllerMessenger.ts

```diff
@@ -82,13 +82,13 @@ export class RestrictedControllerMessenger<
   }: {
     controllerMessenger: ControllerMessenger<Action, Event>;
     name: N;
-    allowedActions: AllowedAction[];
-    allowedEvents: AllowedEvent[];
+    allowedActions?: AllowedAction[] | never;
```

Comment:
per discussion, with `?:`, is `| never` necessary or vice versa?

Reply:
?: allows for null or undefined. 

I believe this is stating that if it is defined it *must* be an `AllowedEvent[]` or it'll throw.

Reply:
sounds reasonable to me 👍 I wasn't entirely sure of the exact behavior with `never`

Reply:
Leaving the `| never` out also serves to ensure that this parameter _must_ be an `AllowedEvent[]` though. I think it's superfluous.

I propagated it here because I used it in `getRestrictedControllerMessenger`, where I needed the type for `AllowedEvents` to be _something_ because optional generic parameters aren't a thing that exists. So I need the `| never` for the generic parameters, but I'll remove them for the optional parameters (here and in `getRestrictedControllerMessenger`)